### ISSUE TITLE
Option for setting instanceId.

### ISSUE THREAD

**Issue by mogensen** [jquatier/eureka-js-client#42]
We have the need to run multiple instances of the same service on the same host machine.
This is not possible then the InstanceId field is automatically defined using the hostname.

I would like the option to set the instanceId field in the configuration array for the Eureka client.

**Comment by kesne** [comment]
Does it sound reasonable to set it under `instance.id`? Then we would pick the `instanceId` by first checking if that is defined, before defaulting to the `hostname`.

Looping in @jquatier for eureka knowledge.

**Comment by jquatier** [comment]
Yeah, I think the question would be whether this is in a AWS environment or not, and what should we do in that scenario. Currently, if its AWS it will pull the instance ID from the AWS metadata. I guess to cover both AWS an non-AWS you would want to do something like this:

```javascript
get instanceId() {
    if(this.config.instance.id) {
      return this.config.instance.id;
    } else if (this.amazonDataCenter) {
      return this.config.instance.dataCenterInfo.metadata['instance-id'];
    }
    return this.config.instance.hostName;
  }
```

**Comment by mogensen** [comment]
Yes. That that would solve the problems we have. :+1:

**Comment by kesne** [comment]
I'll write up a PR later today for this.

**Comment by jquatier** [comment]
The option is available in v2.3.0 of the module. Update your `package.json` and try it out. Thanks @kesne for getting this in.
